presto-native-execution/presto_cpp/main/operators/ShuffleInterface.h

```diff
@@ -37,6 +48,20 @@ class ShuffleInterface {
   /// to be read while noMoreData signals the shuffle service that there
   /// is no more data to be writen.
   virtual bool readyForRead() const = 0;
+
+  static std::unordered_map<std::string, ShuffleInterfaceFactory>& factories() {
+    static std::unordered_map<std::string, ShuffleInterfaceFactory> factories;
+    return factories;
+  }
+
+  static bool registerFactory(
+      const std::string& shuffleInterfaceName,
+      ShuffleInterfaceFactory shuffleInterfaceFactory) {
+    std::unordered_map<std::string, ShuffleInterfaceFactory>& factoryMap =
+        factories();
+    factoryMap.emplace(shuffleInterfaceName, shuffleInterfaceFactory);
+    return true;
```

Comment:
Do you allow for replacing an exiting factory? If so, perhaps, return a boolean indicating whether new entry was created or existing was replaced.